src/main/java/org/sagebionetworks/web/client/widget/table/modal/fileview/EntityViewScopeWidget.java

```diff
@@ -97,28 +97,40 @@ public void configure(EntityBundle bundle, boolean isEditable) {
 		this.isEditable = isEditable;
 		this.bundle = bundle;
 		boolean isVisible = false;
-		if (bundle.getEntity() instanceof EntityView || bundle.getEntity() instanceof Dataset) {
+		if (bundle.getEntity() instanceof EntityView) {
 			isVisible = true;
-			List<Reference> references = new ArrayList<>();
-			if (bundle.getEntity() instanceof EntityView) {
-				currentView = (EntityView) bundle.getEntity();
-				references = getReferencesFromIdList(((EntityView) currentView).getScopeIds());
-				if (tableType == null) {
-					synAlert.consoleError("View type mask is not supported by web client, blocking edit." + ((EntityView) currentView).getViewTypeMask());
-				}
-			} else if (bundle.getEntity() instanceof Dataset) {
-				currentView = (Dataset) bundle.getEntity();
-				references = getReferencesFromDatasetItems(((Dataset) currentView).getItems());
-			}
+
+			currentView = (EntityView) bundle.getEntity();
 			tableType = TableType.getTableType(currentView);
-			viewScopeWidget.configure(references, false, tableType);
-			view.setEditButtonVisible(isEditable && tableType != null);
+			List<Reference> references = getReferencesFromIdList(currentView.getScopeIds());
+
+			// The mask may not be editable since not all masks are supported in the web client
+			boolean canEditMask = isEditable && isEditMaskSupportedInWebClient(tableType);
+			if (isEditable && !isEditMaskSupportedInWebClient(tableType)) {
+				synAlert.consoleError("View type mask is not supported by web client, blocking edit. Mask value:" + ((EntityView) currentView).getViewTypeMask());
+			}
+			view.setEditMaskAndScopeButtonVisible(canEditMask);
+
+			// The scope and mask are both editable from the 'Edit Mask' dialog
+			// The scope is always editable if the user has access, but the button is unnecessary if the user can edit the mask.
+			boolean showEditScopeButton = isEditable && !canEditMask;
+			viewScopeWidget.configure(references, showEditScopeButton, tableType);
```

Comment:
Determines if the user can see the 'Edit Scope' button, which directly opens the entity finder. This button should not be visible if the previous button is visible, because the scope can be edited in the dialog.